Yank/repex.py

```diff
@@ -2099,6 +2215,65 @@ def _attempt_swap(self, replica_i, replica_j):
             self._n_accepted_matrix[thermodynamic_state_i, thermodynamic_state_j] += 1
             self._n_accepted_matrix[thermodynamic_state_j, thermodynamic_state_i] += 1
 
+    # -------------------------------------------------------------------------
+    # Internal-usage: Online Analysis.
+    # -------------------------------------------------------------------------
+
+    @mpi.on_single_node(rank=0, broadcast_result=False, sync_nodes=False)
+    @mpi.delayed_termination
+    @mmtools.utils.with_timer('Computing online free energy estimate')
+    def _run_online_analysis(self):
+        # Prep the reporter
+        self._reporter.close()
+        self._reporter.open('r')
```

Comment:
Hmm. I had it in my head that a dataset open in 'w' mode could not be read from. That was wrong. I have removed those lines